apps/dashboard/src/data/projects/actions.js

```diff
@@ -21,13 +22,21 @@ export function updateProject( projectId, project ) {
 
 export function* saveAndUpdateProject( projectId, project ) {
 	try {
+		yield fetchOngoing();
 		const response = yield saveProject( projectId, project );
 		const id = projectId || response.data.id;
 
-		return updateProject( id, {
+		yield updateProject( id, {
 			...project,
 			id,
 		} );
+
+		if ( ! projectId ) {
+			// this should only happen when creating a new project
+			yield redirect( `/${ id }` );
+		}
+
+		return fetchIdle();
 	} catch ( error ) {
 		// Save failed
```

Comment:
1st of all, if the request fails we're now hanging in a permanent `saving = true` state. You'll need to add a `PROJECT_SAVE_FAILED` / `PROJECT_SAVE_ERROR` / or alike action to handle that case too.

Might as well rename the fetch idle to `PROJECT_SAVE_SUCCESS` then.